File: PlacementicsBackend/src/main/java/com/example/placementicsbackend/services/MentorMenteeService/MentorMenteeService.cs
//Here we going to use Google Calendar API to create, delete, update mentor mentee meetings
using PlacementCellBackend.Services.MentorMenteeService.Interface;
using PlacementCellBackend.Data;
using PlacementCellBackend.DTOs.MentorMeetingDTOs;

namespace PlacementCellBackend.Services.MentorMenteeService;

public class MentorMenteeService : IMentorMenteeService
{
    private readonly AppDbContext _context;
    private readonly GoogleCalendarHelper _calendarHelper;

    public MentorMenteeService(AppDbContext context, GoogleCalendarHelper calendarHelper)
    {
        _context = context;
        _calendarHelper = calendarHelper;
    }

    public async Task<MeetingResponse?> CreateMentorMenteeMeeting(CreateMeetingRequest request)
    {
        try
        {
            var (eventId, meetLink) = await _calendarHelper.CreateMeetingWithMeetLinkAsync(
                request.Title,
                request.Description,
                request.StartTime,
                request.EndTime,
                request.StudentEmail,
                request.AlumniEmail
            );

            return new MeetingResponse
            {
                MeetingId = Guid.NewGuid().ToString(),
                GoogleEventId = eventId,
                GoogleMeetLink = meetLink,
                StartTime = request.StartTime,
                EndTime = request.EndTime
            };
        }
        catch (Exception ex)
        {
            Console.WriteLine($"Error creating meeting: {ex.Message}");
            return null;
        }
    }

    public async Task<bool> DeleteMentorMenteeMeeting(string eventId)
    {
        try
        {
            return await _calendarHelper.DeleteMeetingAsync(eventId);
        }
        catch (Exception ex)
        {
            Console.WriteLine($"Error deleting meeting: {ex.Message}");
            return false;
        }
    }

    public async Task<bool> UpdateMentorMenteeMeeting(UpdateMeetingRequest request)
    {
        try
        {
            return await _calendarHelper.UpdateMeetingAsync(
                request.EventId,
                request.NewTitle,
                request.NewStartTime,
                request.NewEndTime
            );
        }
        catch (Exception ex)
        {
            Console.WriteLine($"Error updating meeting: {ex.Message}");
            return false;
        }
    }
}

File: PlacementicsBackend/src/main/java/com/example/placementicsbackend/services/mentormeentee/impl/MentorMenteeService.java
package com.example.placementicsbackend.services.mentormeentee.impl;

import com.example.placementicsbackend.dtos.MentorMeetingDTOs.MeetingResponse;
import com.example.placementicsbackend.dtos.MentorMeetingDTOs.UpdateMeetingRequest;
import com.example.placementicsbackend.services.mentormeentee.interfaces.IMentorMenteeService;
import java.time.LocalDateTime;
import java.util.UUID;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class MentorMenteeService implements IMentorMenteeService {

    private final GoogleCalendarService googleCalendarService;

    @Override
    public MeetingResponse createMentorMenteeMeeting(
            String title,
            String description,
            LocalDateTime startTime,
            LocalDateTime endTime,
            String studentEmail,
            String alumniEmail) {
        try {
            return googleCalendarService.createMeetingWithMeetLink(
                    title, description, startTime, endTime, studentEmail, alumniEmail);
        } catch (Exception ex) {
            System.out.println("Error creating meeting: " + ex.getMessage());
            return null;
        }
    }

    @Override
    public boolean updateMentorMenteeMeeting(UpdateMeetingRequest request) {
        try {
            return googleCalendarService.updateMeeting(
                    request.getEventId(),
                    request.getNewTitle(),
                    request.getNewStartTime(),
                    request.getNewEndTime());
        } catch (Exception ex) {
            System.out.println("Error updating meeting: " + ex.getMessage());
            return false;
        }
    }

    @Override
    public boolean deleteMentorMenteeMeeting(String eventId) {
        try {
            return googleCalendarService.deleteMeeting(eventId);
        } catch (Exception ex) {
            System.out.println("Error deleting meeting: " + ex.getMessage());
            return false;
        }
    }
}
